platform/macos/os_macos.mm

```diff
@@ -47,17 +47,69 @@
 #include <sys/sysctl.h>
 
 void OS_MacOS::pre_wait_observer_cb(CFRunLoopObserverRef p_observer, CFRunLoopActivity p_activiy, void *p_context) {
-	// Prevent main loop from sleeping and redraw window during modal popup display.
-	// Do not redraw when rendering is done from the separate thread, it will conflict with the OpenGL context updates.
-
-	DisplayServerMacOS *ds = (DisplayServerMacOS *)DisplayServer::get_singleton();
-	if (get_singleton()->get_main_loop() && ds && !get_singleton()->is_separate_thread_rendering_enabled() && !ds->get_is_resizing()) {
-		Main::force_redraw();
-		if (!Main::is_iterating()) { // Avoid cyclic loop.
-			Main::iteration();
+	OS_MacOS *os = static_cast<OS_MacOS *>(OS::get_singleton());
+
+	if (os->run_stage == APP_RUN_STAGE_SETUP) {
```

Comment:
I don't think `APP_RUN_STAGE_SETUP` is needed, as you could move this block of code to the delegate method  [`applicationWillFinishLaunching:`](https://developer.apple.com/documentation/appkit/nsapplicationdelegate/applicationwillfinishlaunching(_:)?language=objc)

Reply:
Yes, probably.